### PR TITLE
Adding support for Sentinel-1 SLCs

### PR DESCRIPTION
**Pull request by constantinius**
This PR aims to provide support for Sentinel-1 SLCs

Since these products differ in some ways from GRD or other product types, they are placed under their own namespace.

In order to reduce the surface size, it extracts commonalities with GRDs and creates subclasses for product discovery and metadata extraction.

Initial implementation work done by @peterfriedl

### CODE REVIEW

**Comment by constantinius** [comment]
@gadomski did you have a chance to look at this? Do you know why the codecov check is failing?

File: examples/slc/S1A_IW_SL1__1_SH_20141031T095929_20141031T100002_003072_003842/S1A_IW_SL1__1_SH_20141031T095929_20141031T100002_003072_003842.json [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+    "s1:total_slices": "10",
+    "s1:swaths": [
+      "IW1",
+      "IW2",
+      "IW3"
+    ],
+    "s1:shape": [
+      21032,
+      14733
+    ],
```
Comment (gadomski) [inline examples/slc/S1A_IW_SL1__1_SH_20141031T095929_20141031T100002_003072_003842/S1A_IW_SL1__1_SH_20141031T095929_20141031T100002_003072_003842.json:76]: This is a duplicate of `proj:shape`, so should be removed IMO?

File: src/stactools/sentinel1/slc/product_metadata.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+                ".//s1sarl1:productTimelinessCategory"
+            ),
+            # "s1:processing_level": self.product_id.split("_")[3][0],
+            # "s1:resolution": resolutions[self.resolution],
+            "s1:orbit_source": self.orbit_source(),
+            "s1:slice_number": self._root.find_text(".//s1sarl1:sliceNumber"),
+            "s1:total_slices": self._root.find_text(".//s1sarl1:totalSlices"),
+            "s1:swaths": [elem.text for elem in self._root.findall(".//s1sarl1:swath")],
+            # "s1:frame_number": get_frame_number(root),
+            # "s1:stop_anxtime": get_stop_anxtime(root),
```
Comment (gadomski) [inline src/stactools/sentinel1/slc/product_metadata.py:39]: If these are unused, they should be removed IMO -- personal preference, but I generally don't like keeping commented-out lines around if they're not required.

**Review by gadomski — CHANGES_REQUESTED** [review]
I've got the codecov fixed, in general looks great! Just a couple little cleanup requests.

File: examples/slc/S1A_IW_SL1__1_SH_20141031T095929_20141031T100002_003072_003842/S1A_IW_SL1__1_SH_20141031T095929_20141031T100002_003072_003842.json (hunk 1 above)
Comment (constantinius) [reply examples/slc/S1A_IW_SL1__1_SH_20141031T095929_20141031T100002_003072_003842/S1A_IW_SL1__1_SH_20141031T095929_20141031T100002_003072_003842.json:76]: This is due to the fact I copied the initial version from GRD. Shall I also fix it there, or do this in a separate PR?

Comment (gadomski) [reply examples/slc/S1A_IW_SL1__1_SH_20141031T095929_20141031T100002_003072_003842/S1A_IW_SL1__1_SH_20141031T095929_20141031T100002_003072_003842.json]: Let's do it in a separate PR, thanks.

**Review by gadomski — APPROVED** [review]

**Pull request merged**